### DISCUSSION TITLE
datatable + modal

### DISCUSSION THREAD

**Discussion by meavric** [livewire/livewire#8607] [7. Bugs (archived)]
### Livewire version

v3.5

### Laravel version

v11

### Which PHP version are you using?

PHP 8.3

### Steps To Reproduce

Hi
In my project I use livewire with datatable and modal. When I show modal:
public function show($regeneration)
{

$this->dispatch('open-modal', 'detailsModal');
}

all add-ons such as search engine, sorting and paging disappear.

When I close modal still all adds are unavilable.

My code:

`<?php

namespace App\Livewire\Ur\Regeneration;

use Livewire\Component;

class RegenerationShowModal extends Component
{

    public $regenerations;

    protected $listeners = ['showRegenerationModal' => 'show'];

    public function show($regeneration)
    {
        
        $this->dispatch('open-modal', 'detailsModal');

    }

    public function close()
    {
        $this->dispatch('close-modal', 'detailsModal');
    }

    public function render()
    {
        return view('livewire.ur.regeneration.regeneration-show-modal');
    }
}
`

and front:
`
<div>
    <x-modal name="detailsModal" title="{{ __('SZCZEGÓŁY') }}" :show="false" maxWidth="2xl">
        <x-slot:body>
            <section class="bg-white dark:bg-gray-900">
                <div class="py-8 lg:py-16 px-4 mx-auto max-w-screen-md">
                    <table id="tabela22" class="table table-light table-striped">
                        <thead>
                            <tr>
                                <th scope="col">{{ __('START') }}</th>
                                <th scope="col">{{ __('STOP') }}</th>
                                <th scope="col">{{ __('STATUS') }}</th>
                                <th scope="col">{{ __('CZAS [min]') }}</th>
                                <th scope="col">{{ __('KTO') }}</th>
                            </tr>
                        </thead>
                        <tbody>
                            {{-- @foreach ($details as $detail)
                            <tr>
                                <td scope="col">{{ $detail->data_start }}</td>
                                <td>{{ $detail->data_stop }}</td>
                                <td>{{ $detail->status }}</td>
                                <td>
                                    @if ($detail->data_stop)
                                    {{ round($detail->data_start->diffInSeconds($detail->data_stop) / 60, 1) }}
                                    @else
                                    N/A
                                    @endif
                                </td>
                                <td>{{ $detail->login }}</td>
                            </tr>
                            @endforeach --}}
                            <x-button-show wire:click="close()">ZAMKNIJ</x-button-show>
                        </tbody>
                    </table>
                </div>
            </section>
            </x-slot>
    </x-modal>

    <table id="tabela" class="table table-light table-striped">
        <thead>
            <tr>
                <th scope="col">{{ __('ID') }}</th>
                <th scope="col">{{ __('Opis') }}</th>
                <th scope="col">{{ __('Zgł. awarii') }}</th>
                <th scope="col">{{ __('Status') }}</th>
                <th scope="col">{{ __('Priorytet') }}</th>
                <th scope="col">{{ __('Data utworzenia') }}</th>
                <th scope="col">{{ __('Action') }}</th>
            </tr>
        </thead>
        <tbody>
            @if (!empty($regenerations))
            @foreach ($regenerations as $regeneration)
            <tr>
                <th scope="row">{{ $regeneration->id }}</th>
                <td>{{ $regeneration->opis }}</td>
                <td>{{ $regeneration->zgloszenie_id }}</td>
                <td>{{ $regeneration->status }}</td>
                <td>{{ $regeneration->prio }}</td>
                <td>{{ $regeneration->created_at }}</td>
                <td>
                    <x-button-show wire:click="show('{{ $regeneration->id }}')">POKAŻ</x-button-show>
                </td>
            </tr>
            @endforeach
            @endif
        </tbody>
    </table>
</div>
`

and modal code:

`@props([
'name',
'show' => false,
'maxWidth' => '2xl',
'title'
])

@php
$maxWidth = [
'sm' => 'sm:max-w-sm',
'md' => 'sm:max-w-md',
'lg' => 'sm:max-w-lg',
'xl' => 'sm:max-w-xl',
'2xl' => 'sm:max-w-2xl',
][$maxWidth];
@endphp

<div x-data="{
        show: @js($show),
        focusables() {
            // All focusable element types...
            let selector = 'a, button, input:not([type=\'hidden\']), textarea, select, details, [tabindex]:not([tabindex=\'-1\'])'
            return [...$el.querySelectorAll(selector)]
                // All non-disabled elements...
                .filter(el => ! el.hasAttribute('disabled'))
        },
        firstFocusable() { return this.focusables()[0] },
        lastFocusable() { return this.focusables().slice(-1)[0] },
        nextFocusable() { return this.focusables()[this.nextFocusableIndex()] || this.firstFocusable() },
        prevFocusable() { return this.focusables()[this.prevFocusableIndex()] || this.lastFocusable() },
        nextFocusableIndex() { return (this.focusables().indexOf(document.activeElement) + 1) % (this.focusables().length + 1) },
        prevFocusableIndex() { return Math.max(0, this.focusables().indexOf(document.activeElement)) -1 },
    }" x-init="$watch('show', value => {
        if (value) {
            document.body.classList.add('overflow-y-hidden');
            {{ $attributes->has('focusable') ? 'setTimeout(() => firstFocusable().focus(), 100)' : '' }}
        } else {
            document.body.classList.remove('overflow-y-hidden');
        }
    })" x-on:open-modal.window="$event.detail == '{{ $name }}' ? show = true : null"
    x-on:close-modal.window="$event.detail == '{{ $name }}' ? show = false : null" x-on:close.stop="show = false"
    x-on:keydown.escape.window="show = false" x-on:keydown.tab.prevent="$event.shiftKey || nextFocusable().focus()"
    x-on:keydown.shift.tab.prevent="prevFocusable().focus()" x-show="show"
    class="fixed inset-0 overflow-y-auto px-4 py-6 sm:px-0 z-50" style="display: {{ $show ? 'block' : 'none' }};">
    <div x-show="show" class="fixed inset-0 transform transition-all" x-on:click="show = false"
        x-transition:enter="ease-out duration-300" x-transition:enter-start="opacity-0"
        x-transition:enter-end="opacity-100" x-transition:leave="ease-in duration-200"
        x-transition:leave-start="opacity-100" x-transition:leave-end="opacity-0">
        <div class="absolute inset-0 bg-gray-500 opacity-75"></div>
    </div>

    <div x-show="show"
        class="mb-6 bg-white rounded-lg overflow-hidden shadow-xl transform transition-all sm:w-full {{ $maxWidth }} sm:mx-auto"
        style="margin-top: 100px;" x-transition:enter="ease-out duration-300"
        x-transition:enter-start="opacity-0 translate-y-4 sm:translate-y-0 sm:scale-95"
        x-transition:enter-end="opacity-100 translate-y-0 sm:scale-100" x-transition:leave="ease-in duration-200"
        x-transition:leave-start="opacity-100 translate-y-0 sm:scale-100"
        x-transition:leave-end="opacity-0 translate-y-4 sm:translate-y-0 sm:scale-95">

        {{-- Modal Body --}}

        @if (isset($title))
        <div class="px-4 py-3 flex items-center justify-between border-b border-gray-300">
            <div class="text-xl text-gray-800">{{ $title }}</div>
            <a href="#" @click="$dispatch('close-modal', '{{ $name }}' )">X</a>
        </div>
        @endif
        <div class="p-4">
            {{ $body }}
        </div>

    </div>
</div>`

### Repository or Wirebox URL

localproject

### Additional information

before
![image](https://github.com/livewire/livewire/assets/56772416/e0525d14-2e3f-4639-bc8c-872d7edf9e92)

when open:
![image](https://github.com/livewire/livewire/assets/56772416/f5c5b8a3-2aca-4a98-8ccf-827137d715fa)

after close:
![image](https://github.com/livewire/livewire/assets/56772416/1d1144ad-2499-48ff-bcb3-b3baccf34f2d)

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by lrljoe** [comment]
Don't make calls from the frontend to a component, just to then dispatch to another component.

Dispatch directly from the client side if you're wanting to dispatch something to another component.

Make sure to wire:key with loops etc, just to help Livewire keep track of elements.

Use forelse, not your wrapping of if then foreach

What errors are you getting in your browser console, there'll be some JS errors I'm certain
